parton/pdf.py

```diff
@@ -182,25 +195,23 @@ def __init__(self, name, member=0, pdfdir=None):
         meta, grids = self.pdfmember.load()
         self.pdfgrids = [PDFGrid.from_block(grid) for grid in grids]
 
-    def xfxQ(self, flavor, x, Q):
+    def xfxQ(self, flavor, x, Q, grid=True):
         """Return x*f(x) by specifying flavor, `x`, and factorization scale
         `Q` in GeV."""
-        return self.xfxQ2(flavor, x, Q**2)
+        return self.xfxQ2(flavor, x, Q**2, grid=grid)
 
-    def xfxQ2(self, flavor, x, Q2):
+    def xfxQ2(self, flavor, x, Q2, grid=True):
         """Return x*f(x) by specifying flavor, `x`, and factorization scale
         squared `Q2` in GeV^2."""
-        res = np.empty((np.size(x), np.size(Q2)))
-        res[:] = np.nan
+        if grid == True:
+            res = np.full((np.size(x), np.size(Q2)), np.nan)
+        else:
+            res = np.full(np.size(x), np.nan)
         for i, pdfgrid in enumerate(self.pdfgrids):
-            _res = np.atleast_2d(pdfgrid.xfxQ2(flavor, x, Q2))
+            _res = pdfgrid.xfxQ2(flavor, x, Q2, grid=grid)
```

Comment:
Why did you remove the `np.atleast_2d`? Is it clear that this will never be needed?

Reply:
When `grid=True` then the output of scipy.interpolate.RectBivariateSpline is always 2D regardless of the shape of the inputs. When `grid=False` anyway the shape will be whatever `np.broadcast_shapes(flavor, x, Q2)` is.